agent/engine/image/types.go

```diff
@@ -37,11 +37,12 @@ func (image *Image) String() string {
 // ImageState represents a docker image
 // and its state information such as containers associated with it
 type ImageState struct {
-	Image      *Image
-	Containers []*api.Container `json:"-"`
-	PulledAt   time.Time
-	LastUsedAt time.Time
-	updateLock sync.RWMutex
+	Image           *Image
+	Containers      []*api.Container `json:"-"`
+	PulledAt        time.Time
+	LastUsedAt      time.Time
+	IsPulledSuccess bool
```

Comment:
`PullSucceeded` seems like a better name for this. Also, please add documentation for this and all other fields.

Reply:
OK, sure.